### PR TITLE
Add gif video from Prismic

### LINKED ISSUE #1991: Add gif-like looping movie content type to Prismic
There is an article in the Yoga series which should feature some gif-like looping videos. We have some of the frontend work for this done already in Cardigan. Now it needs a Prismic content type.

In future we will need to find a way to allow content editors to upload an _actual_ .gif file and automatically create a gif-like video on the fly, but that's out of scope for the current timeframe. That has been stubbed in #1990.

## Acceptance Criteria
- [x] the interface should loop a silent video, emulating a gif
- [x] it should allow for a caption
- [x] it should not play when it's not on screen

## Problem to be solved / User Story
In order to publish gif-videos
As a content editor
I need an appropriate means to do so within Prismic.

### PR DESCRIPTION
**Pull request by davidpmccormick**
Closes #1991 

## Value
Allows content editors to add movies-that-behave-like-gifs to articles.

## Screenshot
gun_show

### All
- [x] PR labelled and assigned
- [x] Any introduced code complexity has been flagged

### UI component only
- [x] A11y tested: `npm run test:accessibility <URL>`
- [x] Browser tested: `npm run test:browsers <URL>`
- [x] Works without JS in the client
- [x] Relevant tracking/monitoring has been considered

### CODE REVIEW

File: client/scss/components/_gif_video.scss [hunk 1]
```diff
@@ -37,6 +37,24 @@
   }
 }
 
+.gif-video__play {
+  .gif-video__play-pause.is-playing & {
+    display: none;
```
Comment (jamesgorrie) [inline client/scss/components/_gif_video.scss]: Could we use the generic hide?
Seems strange to have to write another one?

File: prismic-model/articles.json [hunk 2]
```diff
@@ -69,6 +73,33 @@
...
+                  "label" : "Caption"
+                }
+              },
+              "posterFrame" : {
+                "type" : "Image",
+                "config" : {
+                  "label" : "Poster frame"
+                }
+              },
+              "movieFile" : {
```
Comment (jamesgorrie) [inline prismic-model/articles.json]: Could we use videoFile (and elsewhere where we've used `movie`?) - it's more consistent with HTML etc.

File: server/views/components/gif-video/gif-video.njk [hunk 3]
```diff
@@ -1,18 +1,20 @@
 <figure class="js-gif-video gif-video {{ {s:3} | spacingClasses({margin: ['bottom']}) }}">
   <div class="gif-video__inner">
     <video class="gif-video__video js-gif-video__video"
-      poster="{{ model.posterImage.contentUrl }}" >
-      {% for source in model.sources %}
-        <source src="{{ source.url }}" type='{{ source.fileFormat }}'>
-      {% endfor %}
-      <p class="gif-video__unsupported">Your browser can't handle our GIFs.</p>
+      poster="{{ model.posterFrameUrl }}" >
```
Comment (jamesgorrie) [inline server/views/components/gif-video/gif-video.njk]: Can we set this to a maximum width - not just the original which could be huge?

File: client/scss/components/_gif_video.scss [hunk 4]
```diff
@@ -37,15 +25,20 @@
   }
 }
 
-.gif-video__button-text {
-  @include visually-hidden();
+.gif-video__play {
+  .gif-video__play-pause.is-playing & {
```
Comment (jamesgorrie) [inline client/scss/components/_gif_video.scss]: Why not a modifier? I still don't see when to, and when not to, use BEM?

Comment (jamesgorrie) [reply client/scss/components/_gif_video.scss]: Another question here about what's the point of nesting this? Another thing BEM solves.

File: prismic-model/articles.json (hunk 2 above)
Comment (jamesgorrie) [reply prismic-model/articles.json]: We never really post-fix things with `File` - see `audio` and `image`.

File: client/scss/components/_gif_video.scss (hunk 4 above)
Comment (davidpmccormick) [reply client/scss/components/_gif_video.scss]: As a rule of thumb, I like using `is-blah` classes for state changes brought about via JS – it lends itself to being easier to abstract across JS modules. But given the JS for this module is not in the least bit abstract, I'll bend to your wishes!

**Comment by davidpmccormick** [comment]
re. BEM vs. state-classes – see point 5 here: https://medium.com/fed-or-dead/battling-bem-5-common-problems-and-how-to-avoid-them-5bbd23dee319

**Comment by jamesgorrie** [comment]
>  see point 5 here

>  makes it easy to use javascript to apply generic classes to any component

Yea, I agree with that - but `.is-playing` is not a generic class?

`.is-active`, `.is-visible` might be though. It's just trying to avoid being able to change the style of something somewhere else, which in large codebases, is a very real concern.

But I see it's been changed anywho.

File: server/views/components/captioned-image/captioned-image.njk [hunk 5]
```diff
@@ -11,37 +11,7 @@
...
-        </div>
-        {% if modifiers.full %}
-          <button class="plain-button js-show-hide-trigger">
-            <span class="captioned-image__icon flex flex--v-center flex--h-center bg-transparent-black">
-              {% icon 'information', 'information', ['icon--white'] %}
-            </span>
-          </button>
-        {% endif %}
-      </div>
+      {% componentV2 'tasl', model, modifiers %}
```
Comment (jamesgorrie) [inline server/views/components/captioned-image/captioned-image.njk]: 🍾

**Pull request merged**